src/main/java/org/prebid/server/bidder/readpeak/ReadPeakBidder.java

```diff
@@ -0,0 +1,167 @@
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.bidder.model.Price;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.readpeak.ExtImpReadPeak;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.BidderUtil;
+import org.prebid.server.util.HttpUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class ReadPeakBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpReadPeak>> READPEAK_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private static final String PRICE_MACRO = "${AUCTION_PRICE}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public ReadPeakBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<HttpRequest<BidRequest>> httpRequests = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            final ExtImpReadPeak extImpReadPeak;
+            try {
+                extImpReadPeak = parseImpExt(imp);
+                final Imp modifiedImp = modifyImp(imp, extImpReadPeak);
+                httpRequests.add(makeHttpRequest(request, modifiedImp));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        if (httpRequests.isEmpty()) {
+            return Result.withError(BidderError.badInput("found no valid impressions"));
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private ExtImpReadPeak parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), READPEAK_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Failed to deserialize ReadPeak extension: " + e.getMessage());
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpReadPeak extImpReadPeak) {
+        final Price bidFloorPrice = Price.of(imp.getBidfloorcur(), imp.getBidfloor());
+
+        return imp.toBuilder()
+                .bidfloor(BidderUtil.isValidPrice(bidFloorPrice)
+                        ? bidFloorPrice.getValue()
+                        : extImpReadPeak.getBidFloor())
+                .tagid(extImpReadPeak.getTagId())
+                .build();
```

Comment:
as I can judge the bidFloor logic is incorrect here, we should always put the bidFloor value from the impExt when it's present

regarding the TagId, I believe it makes sense to put it only if it's not blank, otherwise it's better to set null or nothing